### ISSUE TITLE
Issue Credential v2 Credential Preview doesn't allow nested attributes

### ISSUE THREAD

**Issue by TimoGlastra** [hyperledger/aries-rfcs#585]
The [Credential Preview](https://github.com/hyperledger/aries-rfcs/tree/master/features/0453-issue-credential-v2#preview-credential) inner message used in the `propose-credential` and `offer-credential` messages only supports one level of attribute nesting. 

This works fine for indy credentials, as they only support one level of attribute nesting. However the VC Data model allows the credential subject to be any object. The credential preview can't be used to convey a nested attribute structure.

You could use mime-type to show the attribute is a json object, however the value should be a base64 encoded string in this case and suggests to me that you would also include it as a base64 encoded string in the credential itself.

```json
{
    "@type": "https://didcomm.org/issue-credential/%VER/credential-preview",
    "attributes": [
        {
            "name": "person",
            "mime-type": "application/json",
            "value": "<base64 encoded person object>"
        },
    ]
}
```

Looking at present proof v2 the preview message is removed and everything is handled using the attachment format. Is there a specific reason issue credential v2 didn't take this approach?

I think we should either allow the credential preview to support nested attribute structures or remove it and require the attachment format to handle this. 

Thoughts?

**Comment by swcurran** [comment]
I agree this should have been done.  It's a bit tricky what the format should be on an initial message, as I'm assuming the holder should be able to simply propose a credential without proposing the data for the credential, since they don't hold it.  If they are responding to a credential offer with a counter-proposal, they can modify the credential data in the credential offer.

**Comment by TimoGlastra** [comment]
Well I think that is already possible by using the attachment formats. The credential preview is optional, so you just send a credential proposal with only the attachment format. For indy that'll allow you to specify a  schema / cred def / etc.. for the DIF format you can attach a credential manifest.

So I think it should be just about the attributes, conveying everything else in the attachment format.

Easiest format I can think of is to allow for nested `attributes`. This way the old format is still valid and we can update to a `2.1` version. 

```json
{
    "@type": "https://didcomm.org/issue-credential/%VER/credential-preview",
    "attributes": [
        {
            "name": "person",
            "attributes": [
                {
                     "name": "first_name",
                     "value": "John"
                },
                {
                     "name": "last_name",
                     "value": "Smith"
                }
            ]
        },
        {
            "name": "country",
            "value": "NL"
        }
    ]
}
```

As an addition I think we should also the preview to be included in a credential request, as it is allowed for the holder to start from a credential request.

@swcurran I can make a PR with proposed changes as a starting point?

This is gonna be useful for adding JSON-LD credentials to ACA-Py as we don't have to convey the credential data using the attachments in this case

**Comment by swcurran** [comment]
Sounds good -- thanks.
